x-pack/plugins/enterprise_search/public/applications/app_search/types.ts

```diff
@@ -6,3 +6,25 @@
 
 export * from '../../../common/types/app_search';
 export { IRole, TRole, TAbility } from './utils/role';
+
+export interface IApiToken {
+  access_all_engines?: boolean;
+  key?: string;
+  engines?: string[];
+  id?: number;
+  name: string;
+  read?: boolean;
+  type: string;
+  write?: boolean;
+}
+
+export interface IEngine {
+  name: string;
+  type: string;
+  language: string;
+  result_fields: object[];
+}
+
+export interface ICredentialsDetails {
+  engines: IEngine[];
+}
```

Comment:
Super minor note - this interface doesn't seem likely to be reused/shared by other files (unlike IEngine, which is pretty ubiquitous) so I might suggest just keeping it directly in the CredentialsLogic file. Hope that makes sense!

Reply:
One other option is we could also have views/components maintain their own `types.ts` file (similar to how we created a new `credentials/constants.ts` file) if we want to tidy up having misc interface declarations at the top of our logic files!

Reply:
I like your second suggestion

Reply:
https://github.com/elastic/kibana/pull/77626/commits/defa75ffd1aaec041a5bd8a8ab983f66254c265a